example/dcgan.go

```diff
@@ -0,0 +1,222 @@
+package main
+
+import (
+	"compress/gzip"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"os"
+	"os/user"
+	"path"
+
+	torch "github.com/wangkuiyi/gotorch"
+	nn "github.com/wangkuiyi/gotorch/nn"
+	F "github.com/wangkuiyi/gotorch/nn/functional"
+	"github.com/wangkuiyi/gotorch/nn/initializer"
+)
+
+const (
+	images = "train-images-idx3-ubyte"
+	labels = "train-labels-idx1-ubyte"
+	site   = "http://yann.lecun.com/exdb/mnist/"
+)
+
+func downloadMNIST() error {
```

Comment:
The source code for downloading MNIST data duplicates with that in mnist_test.go. Should we extract the code into a new package `gotorch/mnist` to reuse it?

Reply:
Done. Add a new package `gotorch/vision`. 
In Python, there is a torchvision package, it contains all vision dataset related util functions.